docs/getting_started/SERVICES.md

```diff
@@ -0,0 +1,51 @@
+# Services
+
+Bolt provides a concept of _services_ to split up components of Rivet's architecture. See `lib/bolt/config/src/service.rs` to jump straight to the source.
+
+## Service types
+
+### Operation
+
+Simple request/response functions. If an operation fails, it will return the error. (See `GlobalError`.)
+
+These should be used for anything that reads from a database. Anything that's purely logic should be under the `util` library.
+
+Called using the `op!` macro. Under the hood, these are simple function calls, but are abstracted so they can be separate processes communicated using RPC if needed.
+
+### Consumer
+
+Similar to operations, but reads from a durable stream and has no response. If a consumer fails, it will be retried indefinitely.
+
+These should be used for anything that writes to a database.
+
+Messages can be published using the `msg!` macro.
+
+### API
+
+Hosts API endpoints for communication with the outside world.
+
+These should only call operations & publish messages. These should not contain any communication with the database.
+
+### API routes
+
+Routes exposed as a library. These are imported by _API_ services and hosted on the main router.
+
+### Headless
+
+Services that don't allow any incoming network connections. Useful for supporting infrastructure (such as log shipping) or adapting incoming data for Chirp services (such as consuming Nomad events).
```

Comment:
This might be different from what I'm imagining, but to me, "headless" tends to mean "no GUI". Is it being used in that context, but just no networking, like maybe "sandboxed"?

Reply:
Explained it a bit more here fcd61c91. It has networking, so it's not necessarily sandboxed. It's really just a "dumb service." Open to better naming in a separate PR.

Reply:
This makes sense, I'm not sure what best to call this off the top of my head. Will think about it.